src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -509,67 +515,77 @@ public void StopStake()
-                    this.CheckStake(block, chainTip);
+                    this.CheckStake(posBlock, chainTip);
 
                     blockTemplate = null;
                 }
                 else
                 {
-                    this.logger.LogTrace("{0} failed, waiting {1} ms for next round...", nameof(this.StakeAndSignBlockAsync), this.minerSleep);
+                    this.logger.LogTrace("{0} failed to create POS block, waiting {1} ms for next round...", nameof(this.StakeAndSignBlockAsync), this.minerSleep);
                     await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.stakeCancellationTokenSource.Token).ConfigureAwait(false);
                 }
             }
         }
 
+        private async Task<List<UtxoStakeDescription>> GetUtxoStakeDescriptions(WalletSecret walletSecret, CancellationToken ct)
+        {
+            var utxoStakeDescriptions = new List<UtxoStakeDescription>();
+            IEnumerable<UnspentOutputReference> spendableTransactions = this.walletManager
+                .GetSpendableTransactionsInWallet(walletSecret.WalletName, 1)
+                .Where(t => t != null).ToList();
+
+            //todo 
+            FetchCoinsResponse fetchedCoinSet = await this.coinView.FetchCoinsAsync(spendableTransactions.Select(t => t.Transaction.Id).ToArray(), ct).ConfigureAwait(false);
+            if (ct.IsCancellationRequested) return utxoStakeDescriptions;
+
+            foreach (UnspentOutputReference outputReference in spendableTransactions)
+            {
+                UnspentOutputs coinSet = fetchedCoinSet.UnspentOutputs
+                    .FirstOrDefault(f => f.TransactionId == outputReference.Transaction.Id);
+                if (coinSet == null || outputReference.Transaction.Index >= coinSet.Outputs.Length) continue;
+
+                TxOut utxo = coinSet.Outputs[outputReference.Transaction.Index];
+                if (utxo == null || utxo.Value <= MinimumStakingCoinValue) continue;
+
+                uint256 hashBlock = this.chain.GetBlock((int)coinSet.Height)?.HashBlock;
+                if (hashBlock == null) continue;
+
+                var utxoStakeDescription = new UtxoStakeDescription
+                           {
+                               TxOut = utxo,
+                               OutPoint = new OutPoint(coinSet.TransactionId, outputReference.Transaction.Index),
+                               Address = outputReference.Address,
+                               HashBlock = hashBlock,
+                               UtxoSet = coinSet,
+                               Secret = walletSecret // Temporary.
+                           };
+                utxoStakeDescriptions.Add(utxoStakeDescription);
+
+                this.logger.LogTrace("UTXO '{0}' with value {1} might be available for staking.", utxoStakeDescription.OutPoint, utxo.Value);
+
+                if (ct.IsCancellationRequested) return utxoStakeDescriptions;
```

Comment:
not really needed?

Reply:
just to exit a little quicker if we need to, I'm not too fussed honestly

Reply:
I think it would only be needed if there is an expensive operation inside the loop, which I think it is not